test/all.js

```diff
@@ -68,37 +68,35 @@ test('getLink optional params', async t => {
   await serve.ready()
 
   const base = `http://127.0.0.1:${serve.address().port}`
-  t.is(serve.getLink('file', 'an-alias'), `${base}/file?drive=an-alias`)
-  t.is(serve.getLink('file', null, 5), `${base}/file?checkout=5`)
-  t.is(serve.getLink('file', 'an-alias', 5), `${base}/file?drive=an-alias&checkout=5`)
+  t.is(serve.getLink('file', { id: 'an-alias' }), `${base}/file?id=an-alias`)
+  t.is(serve.getLink('file', { version: 5 }), `${base}/file?version=5`)
+  t.is(serve.getLink('file', { id: 'an-alias', version: 5 }), `${base}/file?id=an-alias&version=5`)
 })
 
 test('getLink reverse-proxy usecase', async t => {
   const serve = tmpServe(t, () => {})
   await serve.ready()
 
-  const actual = serve.getLink('file', { protocol: 'https', domain: 'www.mydrive.org' })
-  const expected = 'https://www.mydrive.org/file'
+  const actual = serve.getLink('file', { secure: true, host: 'www.mydrive.org' })
+  const expected = 'https://www.mydrive.org:' + serve.address().port + '/file'
   t.is(actual, expected)
 })
 
 test('getLink reverse-proxy usecase with id', async t => {
   const serve = tmpServe(t, () => {})
   await serve.ready()
 
-  const actual = serve.getLink('file', 'myId', { protocol: 'https', domain: 'www.mydrive.org' })
-  const expected = 'https://www.mydrive.org/file?drive=myId'
+  const actual = serve.getLink('file', { id: 'myId', secure: true, host: 'www.mydrive.org' })
+  const expected = 'https://www.mydrive.org:' + serve.address().port + '/file?id=myId'
```

Comment:
@HDegroote here I changed it because `getLink` can just always return the port in the link, i.e. user doesn't care about this URL so it can always include the port, right?

Reply:
If you want to keep previous behaviour I can revert it, it's an easy change so it's ok

Reply:
Also because it makes the options simpler i.e. `opts.port` always work now, before it was working conditionally based on `opts.host`

Reply:
For the reverse proxy usecase, including the port is bad, because the reverse proxy is listening on another port.

The intended contract of the opts was:
- If you specify a domain, you can specify a port and a protocol if you want to. The default port is no port, and the default protocol is http
- If you do not specify a domain, we just return the address of the server and ignore all other opts (I can't think of a usecase where you would want to change the port there, but fine with adding that if we think of one)

A simple example setup to illustrate the problem:
- serve-drive is running on port 10000, behind a firewall
- I setup a reverse proxy, mapping port 9000 to 10000, and port 9000 is not behind a firewall
- I map domain mydrive.org to port 9000 of that server

=> When I ask for a URL, I expect to get mydrive.org/file?drive=myId, which would be reverse-proxied to mydrive.org:10000/file?drive=myId. If I get mydrive.org:10000?drive=myId then I can't reach that.

But it's a bit more complex still, because the port where both your reverse proxy and serve-drive are running can change over time (for example if serve-drive crashes and it's setup to auto-spin-up a new instance, or if I change my DNS record). All that is covered by the reverse-proxy contract: your interface is mydrive.org, and when talking with it you don't need to worry about ports. So I made getLink behave the same

Reply:
Ok, I understand your case there, but how are you using `getLink` or where? Like you're showing to the user the link from `getLink`?

Will revert the test when I get to the computer

Reply:
But if you add an endpoint then user already knows the link, I think you should assume user knows your API link. In this case, `getLink` is useful for internal usage (apps, systems, etc) so other libs or things use that link but no users

So assume actual users would already know the URL and how the HTTP API works

Reply:
Btw host fixed now!

Reply:
With the adding-an-endpoint solution, the user does not need to know the link. The flow would be:
- get link: mydrive.org/link/my_path_to_file/
- Reverse proxy redirects to wherever servedrive is running, and adds { domain: 'mydrive.org' } to the body
- User gets the link in function of mydrive.org, so mydrive.org/my_path_to_file, and GETs it
- Reverse proxy redirects to servedrive, and returns the file

But this usecase is still hypothetical, so if it simplifies matters, I'm fine with removing that functionality until the need is more concrete. It might very well be easier to have the frontend/app generate the links as you say

Reply:
The problem is that it looks repetitive because from `mydrive.org/link/my_path_to_file/` you go to `mydrive.org/my_path_to_file` so no much benefit for that

If all this is for the users, I think what you actually want is what Mathias said at the very beginning which is this:
https://github.com/mafintosh/directory-index-html

I might add that haha, and we can serve some drives with public domains!